52.md

```diff
@@ -0,0 +1,130 @@
+
+## Calendar Event RSVP
+
+A calendar event RSVP is a response to a calendar event to indicate a user's attendance intention.
+
+If a calendar event tags a pubkey, that can be interpreted as the calendar event creator is inviting that user to be an attendee. Clients MAY choose to prompt the user to RSVP for the calendar event.
+
+Any user may RSVP, even if they were not tagged on the calendar event. Clients MAY choose to prompt the calendar event creator to invite the user who RSVP'd. Clients also MAY choose to ignore these RSVPs.
+
+This NIP is intentionally not defining who is authorized to attend a calendar event if the user who RSVP'd has not been tagged. It is up to the calendar event creator to determine the semantics.
+
+### Format
+
+The format uses a parameterized replaceable event kind `31924` with a list of tags as described below:
+* `a` (required) reference tag to kind `31923` calendar event being responded to per [NIP-33](33.md)
+* `d` (required) universally unique identifier. Generated by the client creating the calendar event RSVP.
+* `t` (required) attendance status. It can be one of the following values: `yes`, `no`, `maybe`
+* `t` (optional) availability status. It can be one of the following values: `free`, `busy`
+* `note` (optional) free-form note that adds more context to this calendar event response.
+
+```json
+{
+  "id": <32-bytes lowercase hex-encoded sha256 of the the serialized event data>,
+  "pubkey": <32-bytes lowercase hex-encoded public key of the event creator>,
+  "created_at": <unix timestamp in seconds>,
+  "kind": "31924",
+  "tags": [
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"],
+    ["d", "<UUID>"]
+
+    ["t", "<yes/no/maybe>"],
+    ["t", "<free/busy>"],
+
+    ["note", "<note>"]
+  ]
+}
+```
+
+## Calendar
+
+A calendar is a collection of calendar events, represented by replaceable list event using kind `30001` via [NIP-51](51.md). A user can potentially have multiple calendars. One may choose to create a calendar to segment calendar events for specific purposes. e.g. personal, work, travel, meetups, conferences.
+
+### Format
+
+The format uses a categorized bookmarks list of kind `30001` with a list of tags as described below:
+* `d` (required) calendar name
+* `a` (repeated) reference tag to kind `31923` calendar event being responded to per [NIP-33](33.md)
+* `t` value set to `calendar` to denote that it is a calendar for easy tag queries.
+
+```json
+{
+  "kind": 30001,
+  "tags": [
+    ["d", "<calendar name>"],
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"],
+    ["a", "31923:<calendar event author pubkey>:<d-identifier of calendar event>", "<optional relay url>"],
+    ["t", "calendar"]
+  ]
+}
+```
+
+## Unsolved Limitations
+
+* No private events
+* No support for time zones
+* No support for recurring events
```

Comment:
I think there's a case to be made to say recurring events just aren't supported. It adds a lot of complexity around RSVP'ing on individual instances, cancelling individual instances, modifying individual instances as one-offs.

Reply:
yes, we also don't need timezones.

Reply:
> Recurring events are a shitcoin and you don't need them for your use case, right?
> 
> I think it's good mentioning they are not supported in this version (hopefully never) and roll with it

I don't need them for my conference app use case, no. That's good with me. Don't want to open this can of worms.